coordinator/ensemble/anti_affinity_selector.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2025 StreamNative, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ensemble
+
+import (
+	"fmt"
+
+	"github.com/emirpasic/gods/sets/hashset"
+	"github.com/pkg/errors"
+
+	"github.com/streamnative/oxia/coordinator/model"
+	p "github.com/streamnative/oxia/coordinator/policies"
+)
+
+var _ Selector = &antiAffinitiesSelector{}
+
+type antiAffinitiesSelector struct {
+}
+
+func (z *antiAffinitiesSelector) SelectNew(
+	candidates []model.Server,
+	candidatesMetadata map[string]model.ServerMetadata,
+	nsPolicies *p.Policies,
+	_ *model.ClusterStatus,
+	replicas uint32) ([]model.Server, error) {
+	if nsPolicies == nil {
+		return candidates, nil
+	}
+	antiAffinities := nsPolicies.AntiAffinities
+	if len(antiAffinities) == 0 {
+		return candidates, nil
+	}
+
+	groupingCandidates := z.groupingCandidates(candidates, candidatesMetadata)
+
+	filteredCandidates := hashset.New()
+	for _, antiAffinity := range antiAffinities {
+		for _, label := range antiAffinity.Labels {
+			labelGroupedCandidates := groupingCandidates[label]
+			for _, servers := range labelGroupedCandidates {
+				if len(servers) > 1 {
+					for idx := range servers[1:] { // todo: avoid always choose first
```

Comment:
Will this be resolved by Oxia loadbalance? Or we can simply shuffle the list first?

Reply:
And I think I'm not fully understand why we need to skip the first server. @mattisonchao Could you please add a comment here?